src/components/PacksReadme/PacksReadme.tsx

```diff
@@ -0,0 +1,190 @@
+  const { isDarkTheme } = useColorMode();
+  const { defaultAlgorithm, darkAlgorithm } = theme;
+  const packName = new URLSearchParams(window.location.search).get("pack")
+  useEffect(() => {
+    const importComponent = async () => {
+      try {
+        const module = await import(`../../../docs/docs-content/integrations/${packName}.md`);
+        const PackReadMeComponent = module.default;
+        setMd(<PackReadMeComponent />);
+      }
+      catch (error) {
+        setMd(null);
+      }
+    };
+    importComponent();
+  }, []);
+
+  const packData = useMemo(() => {
+    const { packs } = usePluginData("plugin-packs-integrations") as PacksIntegrationsPluginData;
+    const _packData = packs.filter((pack) => pack.fields.name === packName)[0];
+    if (_packData) {
+      setSelectedVersion(`${_packData.fields.versions[0].title}`);
+      const packDataInfo: PackReadmeProps = {
+        customDescription: _packData.fields.description,
+        packReadme: _packData.fields.readme,
+        versions: _packData.fields.versions,
+        title: _packData.fields.title,
+        logoUrl: _packData.fields.logoUrl,
+        type: _packData.fields.packType,
+        provider: _packData.fields.cloudTypes,
+      };
+      return packDataInfo;
+    }
+    return {
+      customDescription: "",
+      packReadme: {},
+      versions: [],
+      title: "",
+      logoUrl: "",
+      type: "",
+      provider: [],
+    };
+  }, [packName]);
+
+  function versionChange(version: string) {
+    setSelectedVersion(version);
+  }
+
+  function getOptions() {
+    return packData.versions.map((_version) => {
+      return (<Select.Option key={_version.title}>
+        {_version.title}
+      </Select.Option>)
+    });
+  }
+
+  function renderTabs() {
+    let readme = "";
+    if (Object.keys(packData.packReadme).length && md) {
+      const packUid = packData.versions.find((ver) => ver.title === selectedVersion)?.packUid;
+      readme = packUid ? packData.packReadme[packUid] : "";
+      const items = [
+        {
+          label: `Readme`,
+          key: '1',
+          children: (<Markdown children={readme} />),
+        },
+        {
+          label: `Additional details`,
+          key: '2',
+          children: md,
+        },
+      ] as { label: string, key: string, children: JSX.Element }[];
+
+      return (
+        <Tabs defaultActiveKey="1">
+          {items.map((item) => {
+            return (
+              <Tabs.TabPane tab={item.label} key={item.key}>
+                {item.children}
+              </Tabs.TabPane>
+            )
+          })}
+        </Tabs>
+      );
+    } else if (Object.keys(packData.packReadme).length) {
+      const packUid = packData.versions.find((ver) => ver.title === selectedVersion)?.packUid;
+      readme = packUid ? packData.packReadme[packUid as keyof string] : "";
+      return (<Markdown children={readme} />);
+    } else if (md) {
+      return md;
+    } else {
+      return renderEmptyContent();
```

Comment:
I'm mostly in favor of creating a function for reusability.
if it just wraps some content and is not called more than once, then you can safely inline that content in the return

Reply:
else you have to track what a function truly returns